src/processor/snapshot/database.rs

```diff
@@ -0,0 +1,135 @@
+use std::{
+    convert::{Into, TryFrom},
+    path::PathBuf,
+};
+
+use ethers::types::H256;
+use eyre::Result;
+use rocksdb::{Options, DB};
+use thiserror::Error;
+
+use super::types::{SnapshotFactoryDependency, SnapshotStorageLog};
+
+const STORAGE_LOGS: &str = "storage_logs";
+const INDEX_TO_KEY_MAP: &str = "index_to_key_map";
+const FACTORY_DEPS: &str = "factory_deps";
+const METADATA: &str = "metadata";
+
+const LAST_REPEATED_KEY_INDEX: &str = "LAST_REPEATED_KEY_INDEX";
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Error, Debug)]
+pub enum DatabaseError {
+    #[error("no such key")]
+    NoSuchKey,
+}
+
+pub struct SnapshotDB {
+    db: DB,
+}
```

Comment:
I suggest implementing `std::ops::Deref` on this struct since we're only wrapping another one for now.

Could also replace the named field with an unnamed one (pretty nitty, but I think it makes sense in this circumstance):
```suggestion
pub struct SnapshotDB(DB);

impl Deref for SnapshotDB {
    type Target = DB;

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
```